### ISSUE TITLE
High Sierra, Radeon HD7870

### ISSUE THREAD

**Issue by jief666** [vit9696/Lilu#39]
Hi,
Lilu/WhateverGreen works well for my Radeon HD7870 with 10.11 and 10.12.6. With 10.13.4, it doesn't. I'm comparing the debug log and I'm wondering about something.

**On Sierra (where it works) :**
Lilu:  config @ (DBG) init via mac_cred_check_label_update_execve
Lilu:   iokit @ (DBG) OSPrelinkKextCount equals to 221
Lilu:    mach @ (DBG) MachInfo asKernel 1 object constructed
Lilu:    mach @ (DBG) readMachHeader for /System/Library/Caches/com.apple.kext.caches/Startup/kernelcache
Lilu:    mach @ (DBG) readMachHeader got magic BEBAFECA
Lilu:    mach @ (DBG) readMachHeader got magic 706D6F63
Lilu:    mach @ (DBG) decompressing 13841141 bytes (estimated 42143744 bytes) with 6E767A6C compression mode
Lilu:    mach @ (DBG) readMachHeader got magic FEEDFACF
Lilu:    mach @ (DBG) found kernel mach-o header address at <private>
Lilu:    mach @ (DBG) found executable at path: /System/Library/Caches/com.apple.kext.caches/Startup/kernelcache
Lilu:    mach @ (DBG) header processing found TEXT
Lilu:    mach @ (DBG) header processing found LINKEDIT
Lilu:    mach @ (DBG) header processing found SYMTAB
Lilu:    mach @ (DBG) found kernel mach-o header address at <private>
Lilu:    mach @ (DBG) aslr/load slide is 0x18800000
Lilu:   iokit @ (DBG) failed to find IntelGraphicsFixup
Lilu:    mach @ (DBG) found symbol _vinfo at 0x<private> (non-aslr 0x<private>)

**But on High Sierra :**
Lilu:  config @ (DBG) init via mac_cred_check_label_update_execve
Lilu:   iokit @ (DBG) OSPrelinkKextCount equals to 298
Lilu:    mach @ (DBG) MachInfo asKernel 1 object constructed
Lilu:    mach @ (DBG) readMachHeader for /System/Library/Caches/com.apple.kext.caches/Startup/kernelcache
Lilu:    mach @ (DBG) readMachHeader got magic BEBAFECA
Lilu:    mach @ (DBG) readMachHeader got magic 706D6F63
Lilu:    mach @ (DBG) decompressing 25096163 bytes (estimated 71864320 bytes) with 6E767A6C compression mode
Lilu: patcher @ (DBG) invoked at kext loading/unloading
Lilu: patcher @ (DBG) last kext is <private> and its name is com.apple.driver.AppleFIVRDriver
Lilu: patcher @ (DBG) invoked at kext loading/unloading
Lilu: patcher @ (DBG) last kext is <private> and its name is com.apple.driver.IOPlatformPluginFamily
Lilu: patcher @ (DBG) invoked at kext loading/unloading
Lilu: patcher @ (DBG) last kext is <private> and its name is com.apple.driver.IOPlatformPluginLegacy
Lilu: patcher @ (DBG) invoked at kext loading/unloading
Lilu: patcher @ (DBG) last kext is <private> and its name is com.apple.driver.ACPI_SMC_PlatformPlugin
Lilu: patcher @ (DBG) invoked at kext loading/unloading
Lilu: patcher @ (DBG) last kext is <private> and its name is com.apple.iokit.IOGraphicsFamily
Lilu: patcher @ (DBG) caught the right kext at 0xFFFFFF7F97A6F000, invoking handler
Lilu:    mach @ (DBG) aslr/load slide is 0x<private>

On Sierra, after decompressing, a header FEEDFACF is found. Not on High Sierra.
I've look in MachInfo::initFromFileSystem and I must get "found executable at path:" or "couldn't find a suitable executable", right ?
Can the readMachHeader function just crash and stop ?

Thanks.

**Comment by jief666** [comment]
To understand, I added a lot of debug log inside MachInfo::ReadFromMachHeader.
At first I thought it stops in lzvn_decode. But I took that function and test it in a project with on the same data (my kernelcache). Run perfectly.
Then I add more log before the call to lzvn_decode and the function isn't called anymore.
The more log I add, the less far it goes. Looks like the thread is killed after some time. Is that possible ?

**Comment by jief666** [comment]
After the boot verbose, screen stays light grey. I can access computer by ssh.
Sorry, I didn't even think that it could be only the log that it's eaten. Is that "log bug" new to High Sierra ?

**Comment by jief666** [comment]
It doesn't work on 10.13(.0) either.

**Comment by vit9696** [comment]
No, it is not new to High Sierra, it existed since the log command was introduced (10.11 I think).

**Comment by vit9696** [comment]
Well, if I think of it light grey screen might be related to either AGDC (try different macmodels/turn it on) or WindowServer crashes in fact (check /Library/Logs/DiagnosticReports). Most likely it is really some cfg, so you'd better experiment with them.

**Comment by jief666** [comment]
I didn't think about turning it ON !
Thanks for your work and your answer. I'm leaving for a month, so I'll try again later.
